DSpace/dspace-replicate
# Update for compatibility with DSpace 7 Beta 5

This updates dspace-replicate (RTS) to work with DSpace 7 as of Beta 5.

**Branch Recommendation**

Once merged, `main` will become focused on DSpace 7, as is the case today with the dspace/dspace repository. At that point, further 6.x updates should be submitted against the `dspace-replicate-6_x` branch (which should be fast-forwarded to the old `main` `HEAD`).

**Previous Work**

This PR is a **replacement** for https://github.com/DSpace/dspace-replicate/pull/52 and brings dspace-replicate up to date for compatibility with DSpace 7 Beta 5, released in April 2021.

**Summary of Changes**

* Updates code/dependencies to work against the DSpace 7 API as of Beta 5
* Updates DuraCloud version to 7.0.0 and removes some unnecessary exclusions
* Builds with Java 11 (as DSpace 7 now requires)
* Fixes dependency convergence issues with DSpace 7
* Minimizes the use of static method `Curator.curationContext()` to make unit testing more practical. [See details in the commit message](https://github.com/DSpace/dspace-replicate/pull/60/commits/99cbe55367354641c6278552ed0cc0458a9540eb)
* Updates tests to work with DSpace 7 API changes.

**Review Instructions**

To build and integrate with a DSpace 7 install:

* Make sure you're using JDK 11
* Check out this code standalone, and run `mvn clean install`. This will put built artifacts in your local `~/.m2/repository`.
* Follow instructions [here](https://wiki.lyrasis.org/display/DSPACE/ReplicationTaskSuite#ReplicationTaskSuite-InstallationonDSpace6.x) to install it, but put the following in your `additions/pom.xml` instead of what is given in that doc.

```
      <dependency>
         <groupId>org.dspace</groupId>
         <artifactId>dspace-replicate</artifactId>
         <version>7.0-SNAPSHOT</version>
         <exclusions>
            <exclusion>
               <groupId>com.amazonaws</groupId>
               <artifactId>aws-java-sdk-core</artifactId>
            </exclusion>
            <exclusion>
               <groupId>org.apache.httpcomponents</groupId>
               <artifactId>httpmime</artifactId>
            </exclusion>
            <exclusion>
               <groupId>org.apache.tika</groupId>
               <artifactId>tika-core</artifactId>
            </exclusion>
            <exclusion>
               <groupId>org.springframework.security</groupId>
               <artifactId>spring-security-core</artifactId>
            </exclusion>
         </exclusions>
      </dependency>
```
